examples/particle-swarm-optimizer.dx

```diff
@@ -103,13 +103,13 @@ Run it for more iterations and the result should improve.
 Which it indeed does.
 
 :p optimize 50 10 (new_key 0) rosenbrock2 ([-10.0, -10.0],[20.0, 20.0]) (0.5,0.3,0.4)
-> [4.584775, 21.173801]
+> [0.191434, -0.030977]
 
 :p optimize 50 20 (new_key 0) rosenbrock2 ([-10.0, -10.0],[20.0, 20.0]) (0.5,0.3,0.4)
-> [3.05381, 9.366863]
+> [0.191434, -0.030977]
```

Comment:
How come this one is the same as the one above? And so different from what it used to be? It seems sketchy to me?

Reply:
Nice catch!  It's fine though, it repeats answers because not every step improves the best answer.  It just  happened to get stuck for more than 10 steps here.  Both versions get to the optimum in the end.

As a sanity check, here's the old version's output after every step. 
 It also got stuck a lot at the beginning, just not quite for 10 steps in a row:

```
for i:(Fin 100).
  optimize 50 (ordinal i) (new_key 0) rosenbrock2 ([-10.0, -10.0],[20.0, 20.0]) (0.5,0.3,0.4)
[ [-2.761576, 6.131174]
, [4.654347, 20.24735]
, [4.654347, 20.24735]
, [-1.789254, 3.918399]
, [3.954768, 15.232117]
, [4.646433, 21.41761]
, [4.646433, 21.41761]
, [4.646433, 21.41761]
, [4.646433, 21.41761]
, [4.646433, 21.41761]
, [4.584775, 21.173801]
, [4.584775, 21.173801]
, [4.23832, 17.845085]
, [4.23832, 17.845085]
, [4.23832, 17.845085]
, [4.23832, 17.845085]
, [4.23832, 17.845085]
, [3.05381, 9.366863]
, [3.05381, 9.366863]
, [3.05381, 9.366863]
, [3.05381, 9.366863]
, [3.05381, 9.366863]
, [3.05381, 9.366863]
, [3.05381, 9.366863]
, [3.05381, 9.366863]
, [3.05381, 9.366863]
, [2.401438, 5.927165]
, [2.401438, 5.927165]
, [2.401438, 5.927165]
, [2.662014, 7.043649]
, [2.662014, 7.043649]
, [2.662014, 7.043649]
, [2.662014, 7.043649]
, [1.637838, 2.846597]
```

Reply:
Ah, that makes sense. Thanks for checking that!